xc/xc7/yosys/utils.tcl

```diff
@@ -0,0 +1,22 @@
+# Update the CLKOUT[0-5]_PHASE parameter values.
+# Due to the fact that Yosys doesn't support floating parameter values
+# i.e. treats them as strings, the parameter values need to be multiplied by 1000
+# for the PLL registers to have correct values calculated during techmapping.
+proc update_pll_params {} {
+    set temp_file "[pid].tmp"
+    select t:PLLE2_ADV -list -write $temp_file
```

Comment:
Can we avoid writing this file?

Reply:
Maybe "select" should have an option to set the TCL return as a list, rather than writing a file?

Reply:
I didn't find any other solution to this, since `select` doesn't return anything in TCL

Reply:
I'm suggesting we modify yosys to take a flag like `-tcl_return` or something.

Reply:
@tmichalak This issue is still outstanding?

Reply:
I agree this would be better... But isn't changing Yosys something we want to avoid through using plugins?
Getting this change upstreamed can take some time. Do you want to use SymbiFlow's Yosys fork?

Reply:
I'm hopeful this is something that the upstream Yosys folks would quickly accept or offer a solution too.  I'll reach out to Eddie to test the waters

Reply:
So Eddie proposed another idea that we could do that could be done as a plugin, and with further iteration could be pushed upstream.  So rather than directly modify `select` to set the `Tcl_Return`, we could create a new yosys command like `selection_to_tcl` which converts the current selection into a Tcl list and sets the Tcl_Return as expected.

Sounds good?

Reply:
The new command variant sound less intrusive thus more likely for the Yosys guys to accept.

Reply:
> The new command variant sound less intrusive thus more likely for the Yosys guys to accept.

That was the idea.  It is also something we can do as a plugin right now, while also proposing and iterating it upstream.

Reply:
The plugin is ready: see [PR](https://github.com/SymbiFlow/yosys-symbiflow-plugins/pull/29)